### PR TITLE
Use event instead of threaded loop

### PR DESCRIPTION
**Pull request by JericoFX**
**Describe Pull request**
This pr delete a loop and add a low game event, this handle the UpdateWeaponAmmo and quality of the weapon.

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? ++yes++
- Does your code fit the style guidelines? i think
- Does your PR fit the contribution guidelines? ++yes++

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -256,3 +256,25 @@ CreateThread(function()
...
+
+local lastammo = nil
+--Event that trigger every time the player shoot
+--@a1 | table - The amount of NPC that saw the player shooting, sometime the Player is in this table too
+--@a2 | number - The player who shoot (PlayerPedId)
+--@_  | table - Always empty?
+AddEventHandler("CEventGunShot",function(a1,a2,_)
+    local player = PlayerPedId()
+    local ped = a2
+    if player == ped then --have to do this because the event is fired by the amount of people who saw the player shoot, so if 100 npc hear you shoot, this will trigger 100 times
```
Comment (JericoFX) [inline client/main.lua]: i dont know if this is necessary, the code is client side...

**Comment by D4isDAVID** [comment]
Nice PR, I've made a PR to improve upon it (JericoFX#1).

File: client/main.lua [hunk 2]
```diff
@@ -134,28 +134,28 @@ CreateThread(function()
...
+--                 end
+--             end
+--         else
+--             ammoThread = 500
+--         end
+--         Wait(ammoThread)
+--     end
+-- end)
 
 local shootingThread = 500
```
Comment (D4isDAVID) [inline client/main.lua:160]:
```suggestion
local shootingThread = 500
```
Unnecessary comment, we don't need to keep this.

**Review by D4isDAVID — CHANGES_REQUESTED** [review]

**Review by D4isDAVID — APPROVED** [review]

File: client/main.lua (hunk 1 above)
Comment (D4isDAVID) [reply client/main.lua]: It is, at least obviously when not used by an exploiter.
The events used here (`weapons:server:UpdateWeaponAmmo` and `weapons:server:UpdateWeaponQuality`) are already exploitable on their own, so it doesn't matter much if this event is unsafe.

**Comment by Manason** [comment]
Is this PR a refactor only, or is it added a new behavior? It's a bit unclear from the description.

**Comment by D4isDAVID** [comment]
@Manason
This PR refactors code inside of what previously was a loop, into an event, to be more performant.

File: client/main.lua [hunk 3]
```diff
@@ -256,3 +233,21 @@ CreateThread(function()
...
+---Event that is triggered for gunshots.
+---@param witnesses table  array of peds that witnessed the shots
+---@param ped number  the ped that shot the gun
+AddEventHandler("CEventGunShot", function(witnesses, ped)
+    -- The ped that shot the gun must be the player.
+    if PlayerPedId() ~= ped then return end
+    -- This event can be triggered multiple times for a single gunshot,
+    -- so ignore if the first ped in witnesses is not the player ped.
+    -- (it's always first in the array and shows up only on the first event for the gunshot)
+    if witnesses[1] ~= ped then return end
```
Comment (Manason) [inline client/main.lua:246]: Consider combining these two if statements using an 'and' since they have the same outcome.

File: client/main.lua [hunk 4]
```diff
@@ -256,3 +233,21 @@ CreateThread(function()
         Wait(0)
     end
 end)
+
+---Event that is triggered for gunshots.
+---@param witnesses table  array of peds that witnessed the shots
```
Comment (Manason) [inline client/main.lua:238]: What defines a witness?

**Review by Manason — APPROVED** [review]

File: client/main.lua (hunk 4 above)
Comment (D4isDAVID) [reply client/main.lua:238]: I'm think it should be a ped that heard the shot

Comment (Manason) [reply client/main.lua:238]: What defines a ped that "hears" the shot?

Comment (D4isDAVID) [reply client/main.lua:238]: A ped that can react to the shot, e.g. a running ped will witness the shot and run away

Comment (Manason) [reply client/main.lua:238]: Ah. A native GTA behavior, got it.

**Pull request merged**